torchgeo/datasets/enmap.py

```diff
@@ -0,0 +1,360 @@
+        'B149': 1.60902,
+        'B150': 1.62009,
+        'B151': 1.63111,
+        'B152': 1.64207,
+        'B153': 1.65300,
+        'B154': 1.66387,
+        'B155': 1.67470,
+        'B156': 1.68547,
+        'B157': 1.69620,
+        'B158': 1.70687,
+        'B159': 1.71750,
+        'B160': 1.72808,
+        'B161': 1.73860,
+        'B162': 1.74908,
+        'B163': 1.75951,
+        'B164': 1.93914,
+        'B165': 1.94869,
+        'B166': 1.95820,
+        'B167': 1.96766,
+        'B168': 1.97708,
+        'B169': 1.98645,
+        'B170': 1.99579,
+        'B171': 2.00508,
+        'B172': 2.01433,
+        'B173': 2.02354,
+        'B174': 2.03270,
+        'B175': 2.04183,
+        'B176': 2.05092,
+        'B177': 2.05996,
+        'B178': 2.06897,
+        'B179': 2.07793,
+        'B180': 2.08686,
+        'B181': 2.09574,
+        'B182': 2.10459,
+        'B183': 2.11340,
+        'B184': 2.12217,
+        'B185': 2.13090,
+        'B186': 2.13960,
+        'B187': 2.14826,
+        'B188': 2.15688,
+        'B189': 2.16547,
+        'B190': 2.17402,
+        'B191': 2.18253,
+        'B192': 2.19101,
+        'B193': 2.19945,
+        'B194': 2.20786,
+        'B195': 2.21624,
+        'B196': 2.22458,
+        'B197': 2.23289,
+        'B198': 2.24116,
+        'B199': 2.24940,
+        'B200': 2.25761,
+        'B201': 2.26579,
+        'B202': 2.27393,
+        'B203': 2.28204,
+        'B204': 2.29012,
+        'B205': 2.29817,
+        'B206': 2.30619,
+        'B207': 2.31417,
+        'B208': 2.32213,
+        'B209': 2.33005,
+        'B210': 2.33794,
+        'B211': 2.34581,
+        'B212': 2.35364,
+        'B213': 2.36144,
+        'B214': 2.36921,
+        'B215': 2.37695,
+        'B216': 2.38466,
+        'B217': 2.39234,
+        'B218': 2.40000,
+        'B219': 2.40762,
+        'B220': 2.41521,
+        'B221': 2.42278,
+        'B222': 2.43032,
+        'B223': 2.43782,
+        'B224': 2.44530,
+    }
+
+    def __init__(
+        self,
+        paths: Path | Iterable[Path] = 'data',
+        crs: CRS | None = None,
+        res: float | None = None,
+        bands: Sequence[str] | None = None,
+        transforms: Callable[[dict[str, Any]], dict[str, Any]] | None = None,
+        cache: bool = True,
+    ) -> None:
+        """Initialize a new EnMAP instance.
+
+        Args:
+            paths: one or more root directories to search or files to load
+            crs: :term:`coordinate reference system (CRS)` to warp to
+                (defaults to the CRS of the first file found)
+            res: resolution of the dataset in units of CRS
+                (defaults to the resolution of the first file found)
+            bands: bands to return (defaults to all bands)
+            transforms: a function/transform that takes an input sample
+                and returns a transformed version
+            cache: if True, cache file handle to speed up repeated sampling
+
+        Raises:
```

Comment:
Should assert whether bands names are valid and also raise AssertionError for this

Reply:
Let's actually do this in `RasterDataset`.